src/App.js

```diff
@@ -14,11 +15,75 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    listVisibility: false,
+  }
+
+  showList = () => {
+    this.setState(state => ({
+      ...state,
+      listVisibility: true,
+    }));
+  }
+
+  reverseList = () => {
+    this.setState(state => ({
+      ...state,
+      goods: [...state.goods].reverse(),
+    }));
+  };
+
+  sortByAlphabeticalOrder = () => {
+    this.setState(state => ({
+      ...state,
+      goods: [...state.goods].sort(),
+    }));
+  };
+
+  reset = () => {
+    this.setState(state => ({
+      ...state,
+      goods: goodsFromServer,
+    }));
```

Comment:
```suggestion
    this.setState({goods: goodsFromServer});
```
You're resetting your goods. No need in previous state